src/operator/min.ts

```diff
@@ -12,13 +12,13 @@ import { ReduceOperator } from './reduce';
  * @method min
  * @owner Observable
  */
-export function min<T>(comparer?: (x: T, y: T) => T): Observable<T> {
-  const min: typeof comparer = (typeof comparer === 'function')
-    ? comparer
+export function min<T>(comparer?: (x: T, y: T) => number): Observable<T> {
+  const min: (x: T, y: T) => T = (typeof comparer === 'function')
+    ? (x, y) => comparer(x, y) < 0 ? x : y
     : (x, y) => x < y ? x : y;
   return this.lift(new ReduceOperator(min));
 }
 
 export interface MinSignature<T> {
-  (comparer?: (x: T, y: T) => T): Observable<T>;
+  (comparer?: (x: T, y: T) => number): Observable<T>;
```

Comment:
doesn't `MaxSignature` (https://github.com/ReactiveX/rxjs/pull/1923/files#diff-31007d130dd79eb825771f50f6eef837R24) need to be updated as well?